util/chplenv/chpl_llvm.py

```diff
@@ -776,10 +776,13 @@ def filter_llvm_link_flags(flags):
         # LLVM 15 detects libzstd on some systems but doesn't include
         # the -L path from pkg-config (this can happen in a Spack configuration)
         # So, if we have '-lzstd', use pkg-config to get the link flags.
-        if flag == '-lzstd' and sys.platform != "darwin":
+        if flag == '-lzstd':
           import third_party_utils
-          link_bundled_args, link_system_args = (
-              third_party_utils.pkgconfig_get_system_link_args('libzstd'))
+          try:
+            link_bundled_args, link_system_args = (
+                third_party_utils.pkgconfig_get_system_link_args('libzstd'))
+          except:
+            link_system_args = None
```

Comment:
This way of doing it is short, but it could mask errors, I think? E.g. if I add a typo to `pkgconfig_get_system_link_args` we won't get the Python exception shown anymore.

As a result, I'm recommending that we add `third_party_utils.has_pkgconfig` or something along those lines, which do something like run `pkg-config --version` and check the return code. You could use `try_run_command` for that. And then this code would just check that pkg-config is available instead of platform != darwin.